### ISSUE TITLE
-Wpointer-bool-conversion in drivers/net/ethernet/asix/ax88796c_main.c

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#1492]
```
drivers/net/ethernet/asix/ax88796c_main.c:851:24: error: address of array 'ax_local->phydev->advertising' will always evaluate to 'true' [-Werror,-Wpointer-bool-conversion]
        if (ax_local->phydev->advertising &&
            ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~ ~~
```

Patch submitted: https://lore.kernel.org/r/20211025211238.178768-2-nathan@kernel.org/

**Comment by nathanchance** [comment]
Patch accepted: https://git.kernel.org/netdev/net-next/c/971f5c4079ed

**Comment by nathanchance** [comment]
Fixed in next-20211027.
